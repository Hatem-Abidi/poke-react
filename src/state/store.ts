import { configureStore } from "@reduxjs/toolkit";

import { pokemonApi } from "../services/pokemonApi";
import pokemonReduce from "./pokemon/pokemonSlice";
import pokemonsListReducer from "./pokemonsList/pokemonsListSlice";

export const store = configureStore({
  reducer: {
    pokemonsList: pokemonsListReducer,
    pokemon: pokemonReduce,
    [pokemonApi.reducerPath]: pokemonApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(pokemonApi.middleware),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
