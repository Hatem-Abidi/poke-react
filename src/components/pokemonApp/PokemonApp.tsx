import React from "react";
import { useSelector } from "react-redux";

import { RootState } from "../../state/store";
import Pokemon from "../pokemon/Pokemon";
import PokemonsList from "../pokemonsList/PokemonsList";

import "./PokemonApp.scss";

function PokemonApp() {
  const selectedPokemonIndex = useSelector(
    (state: RootState) => state.pokemon.selectedPokemonIndex
  );
  return (
    <div>
      <h1>Pokemon React</h1>
      <div className="pokemon-react__container">
        <div>
          <PokemonsList />
        </div>
        <div>
          <div className="pokemon-react__pokemon-container">
            {selectedPokemonIndex && <Pokemon />}
          </div>
        </div>
      </div>
    </div>
  );
}

export default PokemonApp;
