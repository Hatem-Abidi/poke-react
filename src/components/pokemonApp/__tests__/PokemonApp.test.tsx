import React from "react";
import type { ReactNode } from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";

import PokemonApp from "../PokemonApp";
import { store } from "../../../state/store";

function Wrapper(props: { children: ReactNode }) {
  return <Provider store={store}>{props.children}</Provider>;
}

describe("render PokemonApp", () => {
  it("render PokemonApp", () => {
    const component = render(<PokemonApp />, { wrapper: Wrapper });
    expect(component).toMatchSnapshot();
  });
});
