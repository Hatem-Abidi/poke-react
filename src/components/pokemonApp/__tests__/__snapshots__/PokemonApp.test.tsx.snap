// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render PokemonApp render PokemonApp 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1>
          Pokemon React
        </h1>
        <div
          class="pokemon-react__container"
        >
          <div>
            <div
              class="pokemon-list__container"
            >
              <h2>
                Pokemons List (
                )
              </h2>
              <div>
                Pokemons per page:
                <select
                  class="pokemon-list__pokemons-per-page"
                >
                  <option
                    value="10"
                  >
                    10
                  </option>
                  <option
                    value="20"
                  >
                    20
                  </option>
                  <option
                    value="30"
                  >
                    30
                  </option>
                  <option
                    value="40"
                  >
                    40
                  </option>
                  <option
                    value="50"
                  >
                    50
                  </option>
                </select>
              </div>
              <div
                class="pokemon-list__alert-message"
              >
                Loading...
              </div>
            </div>
          </div>
          <div>
            <div
              class="pokemon-react__pokemon-container"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        Pokemon React
      </h1>
      <div
        class="pokemon-react__container"
      >
        <div>
          <div
            class="pokemon-list__container"
          >
            <h2>
              Pokemons List (
              )
            </h2>
            <div>
              Pokemons per page:
              <select
                class="pokemon-list__pokemons-per-page"
              >
                <option
                  value="10"
                >
                  10
                </option>
                <option
                  value="20"
                >
                  20
                </option>
                <option
                  value="30"
                >
                  30
                </option>
                <option
                  value="40"
                >
                  40
                </option>
                <option
                  value="50"
                >
                  50
                </option>
              </select>
            </div>
            <div
              class="pokemon-list__alert-message"
            >
              Loading...
            </div>
          </div>
        </div>
        <div>
          <div
            class="pokemon-react__pokemon-container"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
